### PR TITLE
Migrate AdService to Gradle 6

### PR DESCRIPTION
**Pull request by elsesiy**
Update build.gradle to use plugin DSL instead of outdated buildscript
Update gradle wrapper to 6.3
Update jackson, opencensus, grpc & other dependencies to latest versions
Fix code warnings
Properly format source files using goJF task
Add default port to AdService

### CODE REVIEW

File: src/adservice/src/main/java/hipstershop/AdService.java [hunk 1]
```diff
@@ -54,14 +55,16 @@
...
   private static int MAX_ADS_TO_SERVE = 2;
+
   private Server server;
   private HealthStatusManager healthMgr;
 
   private static final AdService service = new AdService();
 
   private void start() throws IOException {
-    int port = Integer.parseInt(System.getenv("PORT"));
+    int port = Integer.parseInt(System.getenv().getOrDefault("PORT", "6565"));
```
Comment (rghetia) [inline src/adservice/src/main/java/hipstershop/AdService.java]: should this be 9555?

Comment (elsesiy) [reply src/adservice/src/main/java/hipstershop/AdService.java]: @rghetia Good catch, fixed!

**Comment by elsesiy** [comment]
@rghetia Is there anything missing from your perspective?

**Comment by askmeegs** [comment]
@DanSanche can we merge this? Thanks!

**Comment by elsesiy** [comment]
@askmeegs I just rebased the branch again and resolved some conflicts, it would be great if we can merge it soon 👍 Not sure what's blocking this for almost 3 months now.

**Comment by elsesiy** [comment]
@rghetia @DanSanch Is there anything blocking this from being merged?

**Comment by rghetia** [comment]
> @rghetia @dansanch Is there anything blocking this from being merged?

@ahmetb do you still own this repo? 
I can merge this but it would be better if an active maintainer does this.

File: src/adservice/src/main/java/hipstershop/AdServiceClient.java [hunk 2]
```diff
@@ -97,14 +97,13 @@ public void getAds(String contextKey) {
     }
   }
 
-  private static int getPortOrDefaultFromArgs(String[] args, int index, int defaultPort) {
-    int portNumber = defaultPort;
-    if (index < args.length) {
+  private static int getPortOrDefaultFromArgs(String[] args) {
+    int portNumber = 9555;
+    if (2 < args.length) {
```
Comment (ahmetb) [inline src/adservice/src/main/java/hipstershop/AdServiceClient.java]: does this change the behavior at all? I think we support $PORT env var to be able to also set the port number.

**Comment by ahmetb** [comment]
Yeah I still have write permissions but @DanSanche is the new maintainer.

File: src/adservice/src/main/java/hipstershop/AdServiceClient.java (hunk 2 above)
Comment (elsesiy) [reply src/adservice/src/main/java/hipstershop/AdServiceClient.java]: No it doesn't, the port can still be dynamically set. This just moves the static parameters inside the function.

**Pull request merged**